ClinicalTrial_SE_curation.Rmd

```diff
@@ -0,0 +1,320 @@
+  pheno_data$unique_patient_ID <- rownames(pheno_data)
+  prior <- c("cancer_type","treatment", "response", "survival_type", "survival_time", "survival_time_unit", "event_occured")
+  
+  for (c in prior){
+    if (is.na(get(c))){
+      pheno_data[[c]] <- NA
+    } else {
+      pheno_data[[c]] <- pheno_data [,  get(c)] 
+    }
+  }
+  
+  # Ordering columns based on priority
+  pheno_data <- pheno_data[, c("unique_patient_ID", prior, colnames(pheno_data)[!colnames(pheno_data) %in% prior])]
+  
+  # Replacing "NA" with NA_character_ entries
+  pheno_data[pheno_data=="NA"]=NA
+  
+  
+  # Feature data
+  feat_data <- merge(features_gene, data.frame(gene_id = rownames(assay_data)), all.y=T, by = "gene_id")
+  rownames(feat_data) <- feat_data$gene_id
+  
+  eSet <- ExpressionSet(assayData = as.matrix(assay_data), 
+                        phenoData = AnnotatedDataFrame(pheno_data), 
+                        featureData = AnnotatedDataFrame(feat_data[rownames(assay_data),])) 
+}
+
+```
+
+```{r}
+# ExpressionSet to SummarizedExperiment
+eSetToSE <- function(eSet) {
+  
+  BiocGenerics::annotation(eSet) <- "rna"
+  stopifnot(all(rownames(fData(eSet)) == rownames(exprs(eSet))))
+  stopifnot(all(rownames(pData(eSet)) == colnames(exprs(eSet))))
+  
+  # Build summarized experiment from eSet
+  SE <- SummarizedExperiment::SummarizedExperiment(
+    assays=SimpleList(as.list(Biobase::assayData(eSet))),
+    # Switch rearrange columns so that IDs are first, probes second
+    rowData=S4Vectors::DataFrame(Biobase::fData(eSet)),
+    colData=S4Vectors::DataFrame(Biobase::pData(eSet)),
+    metadata=list("experimentData" = eSet@experimentData, 
+                  "annotation" = Biobase::annotation(eSet), 
+                  "protocolData" = Biobase::protocolData(eSet)))
+  
+  # Extract names from expression set                  
+  SummarizedExperiment::assayNames(SE) <- Biobase::assayDataElementNames(eSet)
+  
+  stopifnot(all(rownames(colData(SE)) == rownames(pData(eSet))))
+  stopifnot(all(rownames(rowData(SE)) == rownames(fData(eSet))))
+  stopifnot(all(rownames(colData(SE)) == colnames(assay(SE))))
+  
+  return(SE)
+}
+```
+
+```{r}
+# Run all the functions
+run_all <- function(raw_tar, geo_path, brain_path, delim, GPL, cancer_type ,treatment, response, survival_type,
+                    survival_time, survival_time_unit, event_occured, delim2=delim2){
+  
+  rma <- geo_rma(raw_tar = raw_tar, 
+                 geo_path = geo_path , 
+                 brain_path= brain_path)
+  
+  GSE <- sub("_.*", "",raw_tar)
+  eset <- ESet(rma.norm = rma, GSE = GSE, delim = delim, GPL = GPL, delim2=delim2, cancer_type=cancer_type ,
+               treatment =treatment, response= response, survival_type= survival_type,
+               survival_time= survival_time, survival_time_unit=survival_time_unit,  event_occured= event_occured)
+  SE <- eSetToSE(eset)
+  
+  return(SE)
+}
+
+```
+
+```{r}
+#==== 1 ====
+GSE41998_SE <- run_all(raw_tar = "GSE41998_RAW.tar" , 
```

Comment:
Suggestion:

It seems like many of the parameters you specified are the default. It is okay to exclude parameters from a function call if they are using the defaults for the function definition. This saves you time typing, and can also help focus on the relevant things being changed, enhancing the clarify of code for the reader.